src/fides/api/alembic/migrations/versions/c9c72b3d550b_data_migration_converting_existing_.py

```diff
@@ -0,0 +1,214 @@
+        )
+    )
+
+    logger.debug("Converting existing cookies to assets")
+    assets_to_create: Dict[str, Asset] = {}
+    for row in result:
+        if row.privacy_declaration_id is None:
+            # If the privacy declaration ID is None, we skip matching for this row
+            assets_to_create[row.name] = Asset(
+                id=str(uuid.uuid4()),
+                created_at=row.created_at,
+                updated_at=row.updated_at,
+                name=row.name,
+                domain=row.domain,
+                system_id=row.system_id,
+                data_uses=[],
+                asset_type="Cookie",
+            )
+            continue
+        pud_system_id_data_use = connection.execute(
+            sa.text(
+                "SELECT system_id, data_use FROM privacydeclaration WHERE id = :privacy_declaration_id"
+            ),
+            privacy_declaration_id=row.privacy_declaration_id,
+        ).fetchone()
+
+        if pud_system_id_data_use:
+            pud_system_id, data_use = pud_system_id_data_use
+        if pud_system_id:
+            system_id = pud_system_id
+        else:
+            system_id = row.system_id
+        if data_use:
+            data_uses = [data_use]
+        else:
+            data_uses = []
+
+        # Asset unique identifier is a composite of name, asset_type, domain, base_url, system_id. All assets here are cookies.
+        identifier = f"{row.name}_{'Cookie'}_{row.domain}_{row.path}_{system_id}"
+        if identifier in assets_to_create.keys():
+            # If the asset already exists, append the data_use to the existing asset
+            assets_to_create[identifier].data_uses.extend(data_uses)
+        else:
+            # If the asset does not exist, create a new asset with the data_use
+            assets_to_create[identifier] = Asset(
+                id=str(uuid.uuid4()),
+                created_at=row.created_at,
+                updated_at=row.updated_at,
+                name=row.name,
+                domain=row.domain,
+                system_id=system_id,
+                data_uses=data_uses,
+                asset_type="Cookie",
+            )
+
+    # Insert the assets into the asset table
+    logger.debug("Inserting assets into asset table ")
+    for asset in assets_to_create.values():
+
+        connection.execute(
+            sa.text(
+                "INSERT INTO asset (id, created_at, updated_at, name, domain, system_id, data_uses, asset_type, with_consent) "
+                "VALUES (:id, :created_at, :updated_at, :name, :domain, :system_id, :data_uses, :asset_type, false) "
+                "ON CONFLICT DO NOTHING"
+            ),
+            id=str(uuid.uuid4()),
+            created_at=asset.created_at,
+            updated_at=asset.updated_at,
+            name=asset.name,
+            domain=asset.domain,
+            system_id=asset.system_id,
+            data_uses=asset.data_uses,
+            asset_type="Cookie",
+        )
```

Comment:
again, this'll issue `n` queries for `n` assets, right? the number of assets we're creating here could be quite large, i think we should really try to get this done in a single bulk query 👍

Reply:
bulk insert here: https://github.com/ethyca/fides/pull/5776/files#diff-9a71bb9fcb8781cd9b038e712d899749d3e99ec8344ce94cfa617e6dbb0785a6R101